src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,218 @@
+package duke;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+/**
+ * Class for TaskList components of Duke.
+ */
+public class TaskList {
+    /**
+     * Class for a singular Task.
+     */
+    class Task {
+        protected String description;
+        protected boolean isDone;
+
+        public Task(String description) {
+            this.description = description;
+            this.isDone = false;
+        }
+
+        public boolean getIsDone() {
+            return isDone;
+        }
+
+        public String getDescription() {
+            return description;
+        }
+
+        public void mark() {
+            this.isDone = true;
+        }
+
+        public void unmark() {
+            this.isDone = false;
+        }
+
+        public String getSavedString() {
+            return String.format("T | %s | %s\n",this.isDone, this.description);
+        }
+
+        @Override
+        public String toString() {
+            return String.format("[T] %s %s", Ui.checkbox(this.getIsDone()), this.getDescription());
+        }
+    }
+
+    /**
+     * Class for a singular Deadline.
+     */
+    class Deadline extends Task {

```

Comment:
Would it be better if the various Task subclasses (e.g. Deadline, Todo) were included in their own individual files?